src/typing/typeloadModule.ml

```diff
@@ -244,15 +244,20 @@ module ModuleLevel = struct
 		in
 		let tdecls = List.fold_left make_decl [] tdecls in
 		let tdecls =
-			match !statics with
-			| [] ->
+			match !statics, m.m_statics with
+			| [], _ ->
 				tdecls
-			| statics ->
+			| statics, Some c ->
+				raise_typing_error_ext (make_error ~sub:[
+					make_error ~depth:1 (Custom "Previously defined here") c.cl_pos
+				] (Custom (Printf.sprintf "Cannot add module level fields for module %s" (s_type_path m.m_path))) loadp);
```

Comment:
The phrasing somehow suggests that this could be allowed for a different module.

Reply:
`"Module level fields are already defined for module %s"` ?

Reply:
Actually this makes me realize that we're not addressing the real problem here. This should be about adding module level fields to existing modules, whether or not they are already defined should not make a difference...

Reply:
Adding module level fields to an existing module that does not have module level fields already works. It's only when it already has module level fields that we're having issues.

Reply:
That doesn't sound right, I'll have to take a closer look at the original issue in that case...

Reply:
Yeah, [repro](https://github.com/HaxeFoundation/haxe/issues/10374) from the issue is actually bad code since it's checking `Context.getType` with a module level field instead of a type, so always running `defineModule`.

It runs fine the first time because Test.hx has no module level fields, but obviously fails the next times because it is redefining the module level fields.